src/main/java/org/truffleruby/core/rubinius/IONodes.java

```diff
@@ -438,6 +444,42 @@ public DynamicObject ensureOpen(DynamicObject file,
 
     }
 
+    @Primitive(name = "io_read_polyglot")
+    public static abstract class IOReadPolyglotNode extends IOPrimitiveArrayArgumentsNode {
+
+        @TruffleBoundary(transferToInterpreterOnException = false)
+        @Specialization
+        public DynamicObject read(int fd, int length,
+                @Cached("create()") MakeLeafRopeNode makeLeafRopeNode) {
+            if (fd != 0) {
+                throw new UnsupportedOperationException();
```

Comment:
I'd really like us to document why an operation is unsupported in the exception message. It saves a lot of mental work for the next person to encounter this.

Reply:
I'll make it an assert with a message, that should be clearer.